### ISSUE TITLE
ENH: odo support

### ISSUE THREAD

**Issue by femtotrader** [dr-leo/pandaSDMX#18]
Hello,

Maybe you should consider to support odo
http://odo.readthedocs.org/

Kind regards

**Comment by dr-leo** [comment]
Hi,

I've quickly looked at odo.
It seems awesome.

Some thoughts and questions:
1. pandaSDMX has native Support for Pandas. From there you can use odo.
2. One could think of odo Support for sdmx files and
pandasdmx.api.Response instances. Both can contain data and all kinds
of metadata. Metadata can be stored anywhere but not in Pandas. Files
can be stored in databases etc. Data can be exported vis Pandas to
other formats such as xslx.
3. Convert functions: Should odo Registration go into
pandasdmx.__init__.py so they are registered on each import?
4. What of the above would make sense?
5. Could you give some implementation help or even send a PR?

Leo

**Comment by femtotrader** [comment]
Hi @dr-leo ,

You might have a look at https://github.com/blaze/odo/issues/317

Kind regards

**Comment by dr-leo** [comment]
Thanks, PhemtoTrader.

Some more thoughts:

1. I now think any odo-related functions should go into api.py. This
seems to require importing Pandas there as well.

2. Scope of odo Support:

a) To begin with, an 'sdmx_to_pandas' function should convert a
pandasdmx.api.Response instance to PD.DataFrame, i.e. a wrapper around
Response.write().

b) One could also consider some 'sdmxml_file_to_pandas' function that
converts an XML-file as received from an SDMX Server into a
PD.DataFrame. Maybe odo's URL's could be leveraged. Passing a file or
filepath to odo would be the low-hanging fruit. But maybe it is useful
to Support sdmxml-files stored in a database, too.

3. Things that will likely be out of scope:

a)	Append: SDMX datasets cannot be appended. At least I don’t know
how. Well, if one makes certain assumptions one could convert both to
PD-DataFrame and append those. But this seems not very interesting,
right?

b)	Discover / shape: An SDMX dataset has no meaningful shape. A
dataset consisting of time series may have various time-periods such
as annual, monthly, daily etc. It is meaningless to speak of a shape
here as the indices cannot be merged etc. Only once a PD.DataFrame has
been created from an SDMX Response it makes sense to calculate the
shape.

4. Finally, I have to say that I won’t have the time needed to dive
into odo deeply enough to do this, at least not in the next six months
or so. If I ever find the time to work on pandaSDMX beyond bug-fixing,
I think I should spend it on a reader for SDMXJSON so that OECD and
IMF SDMX Services are supported.

So it would be phantastic if you could contribute a few lines of code
on odo Support. I would happily do a v0.3.2 release for that and
acknowledge your contribution. I have added another small yet cool API
Extension, namely a __getattr__ method for the api.Response class to
bypass the 'msg' Attribute. So one can now write: 'resp.dataflows'
instead of 'resp.msg.dataflows'.

Leo
